### PR TITLE
Fetch-lib and related upkeep (salavging valuables from #58)

### PR DESCRIPTION
**Pull request by sed-i**
This PR is a bunch of valuable changes salvaged from #58:
- fetch-lib
- mypy updates due to fetch-lib
- smarter config update
- (Renaming of test_charm to test_config_changed_modifies_file shows up as deletion + new file.)

### CODE REVIEW

File: src/charm.py [hunk 1]
```diff
@@ -298,27 +298,65 @@ def _update_config(self) -> None:
...
         config_hash = sha256(config_yaml)
 
-        logger.debug(
-            "config changed" if config_hash != self._stored.config_hash else "no change",
-        )
+        if config_hash == self._stored.config_hash:
+            logger.debug("no change in config")
+            return
+        else:
+            logger.debug("config changed")
```
Comment (Abuelodelanada) [inline src/charm.py:305]: Because we have a `return`, the `else` is not necessary

**Review by Abuelodelanada — APPROVED** [review]

File: src/charm.py (hunk 1 above)
Comment (sed-i) [reply src/charm.py:305]: Removed. Thanks!

File: lib/charms/karma_k8s/v0/karma_dashboard.py [hunk 2]
```diff
@@ -37,7 +37,10 @@ def __init__(self, *args):
 
 # Increment this PATCH version before using `charmcraft publish-lib` or reset
 # to 0 if you are raising the major API version
-LIBPATCH = 1
```
Comment (mmanciop) [inline lib/charms/karma_k8s/v0/karma_dashboard.py:40]: Current patch version on CharmHub is already 3, is this correct?

Comment (sed-i) [reply lib/charms/karma_k8s/v0/karma_dashboard.py:40]: Correct.

Comment (simskij) [reply lib/charms/karma_k8s/v0/karma_dashboard.py:40]: if it's already 3, shouldn't this be 4?

Comment (sed-i) [reply lib/charms/karma_k8s/v0/karma_dashboard.py:40]: This is the alertmanager charm, it does not own the karma lib.
This change is after `fetch-lib`, so yes, it matches charmhub.

**Review by balbirthomas — APPROVED** [review]

**Pull request merged**